plugins/fontdownloader.koplugin/main.lua

```diff
@@ -0,0 +1,535 @@
+        ["tamil"]               = _("Tamil"),
+        ["telugu"]              = _("Telugu"),
+        ["thai"]                = _("Thai"),
+        ["tibetan"]             = _("Tibetan"),
+        ["vietnamese"]          = _("Vietnamese"),
+    },
+    variants = {
+        ["regular"]             = "regular",
+        ["italic"]              = "italic",
+        ["500"]                 = "medium",
+        ["500italic"]           = "medium-italic",
+        ["700"]                 = "bold",
+        ["700italic"]           = "bold-italic",
+    },
+}
+
+function FontDownloader:init()
+    self.ui.menu:registerToMainMenu(self)
+    local keyfile = io.open(self.user_key, "r")
+    if keyfile then
+        local key = keyfile:read("*a")
+        keyfile:close()
+        self.api_key = key
+        return
+    end
+    self.api_key = "AIzaSyDQZaihK8Lb7jJ3DYyrQrpyyJF7tyvrqAs"
+end
+
+function FontDownloader:addToMainMenu(menu_items)
+    -- plugin settings
+    menu_items.font_downloader = {
+        text = _("Font downloader"),
+        sub_item_table = {
+            {
+                text = _("Set download location"),
+                callback = function()
+                    self:setFontDir()
+                end,
+            },
+            {
+                text = _("Sync font index"),
+                callback = function()
+                    self.fonts = self:fontTable()
+                end,
+            },
+        }
+    }
+    -- inject in font menu for CRE documents
+    if self.ui.document and self.ui.document.provider == "crengine" then
+        local menu_orig = menu_items.change_font.sub_item_table[1].sub_item_table
+        local position = #menu_orig - 1 -- above fallback fonts
+        local menu = menu_orig
+        table.insert(menu, position, {
+            text = _("Download more fonts"),
+            callback = function()
+                self:frontpage()
+            end,
+        })
+        menu_items.change_font.sub_item_table[1].sub_item_table = menu
+    end
+end
+
+function FontDownloader:onFontDownloadLookup()
+    self:frontpage()
+    return true
+end
+
+function FontDownloader:onNetworkConnected()
+    if self.pending_action then
+        -- execute pending actions
+        self.pending_action()
+        self.pending_action = nil
+    end
+end
+
+-- frontpage for font lookup
+--
+-- the function schedules itself as a pending action if it fails for some reason,
+-- so we keep a retry count to avoid "fail" loops
+function FontDownloader:frontpage(retry_count)
+    if retry_count and retry_count > 3 then
+        logger.warn("Too many retries. Giving up")
+        return
+    end
+
+    -- returns true if modification time is equal or less than 7 days old
+    local ok_stamp = function(timestamp)
+        if not timestamp then return false end
+        local current = os.date(self.timestamp_format)
+        if tonumber(current) > timestamp + 7 then return false end
+        return true
+    end
```

Comment:
```lua
if tonumber(current) > timestamp + 7 then return false end
```
Is this conversion correct? For example:
`current = 20200701` and `timestamp = 20200630`
I assume the comparison would look like `current(20200701) > timestamp(20200637)`. This would return false, when in fact it should be true

Reply:
Good catch!